backend/fms_core/models/sample.py

```diff
@@ -62,6 +64,10 @@ def is_pool(self) -> bool:
     def derived_sample_not_pool(self) -> DerivedSample:
         return self.derived_samples.first() if not self.is_pool else []  # Forces crash if pool
 
+    @property
+    def biosample(self) -> Biosample:
+        return self.derived_sample_not_pool.biosample if not self.is_pool else None
```

Comment:
Do not use derived_sample_not_pool inside generic properties, the way we deal with pool is likely to be context specific and wrapping this in here will prevent this.